datafusion/core/tests/order_spill_fuzz.rs

```diff
@@ -118,7 +118,7 @@ fn make_staggered_batches(len: usize) -> Vec<RecordBatch> {
             RecordBatch::try_from_iter(vec![(
                 "x",
                 Arc::new(Int32Array::from_iter_values(
-                    std::iter::from_fn(|| Some(rng.gen())).take(to_read),
+                    (0..to_read).map(|_| rng.gen()),
```

Comment:
This seemingly innocuous change is necessary because the previous formulation results in a size hint of 0, resulting in bump allocation of the Int32Array. This in turn causes it to be larger than it should be. Previously as we sorted on ingest this wasn't a problem, as it would rewrite the array, we no longer do this and so the memory accounting is impacted by this

Reply:
I recommend a comment explaining this

Reply:
It seems strange to explain what was previously just an implementation bug, I think it is fine?

Reply:
I guess I was thinking that without a comment it could easily regress in the future 🤷  maybe the memory accounting failures would catch it